xonsh/tools.py

```diff
@@ -346,6 +348,44 @@ def subproc_toks(line, mincol=-1, maxcol=None, lexer=None, returnline=False):
     return rtn
 
 
+def check_bad_str_token(tok):
+    """Checks if a token is a bad string."""
+    if tok.type == 'ERRORTOKEN' and tok.value == 'EOF in multi-line string':
+        return True
+    elif isinstance(tok.value, str) and not check_quotes(tok.value):
+        return True
+    else:
+        return False
+
+
+def check_quotes(s):
+    """Checks a string to make sure that if it starts with quotes, it also
+    ends with quotes.
+    """
+    starts_as_str = RE_BEGIN_STRING.match(s) is not None
+    ends_as_str = s.endswith('"') or s.endswith("'")
+    if not starts_as_str and not ends_as_str:
+        ok = True
+    elif starts_as_str and not ends_as_str:
+        ok = False
+    elif not starts_as_str and ends_as_str:
+        ok = False
+    else:
+        m = RE_COMPLETE_STRING.match(s)
+        ok = m is not None
+    return ok
+
+
+def _have_open_triple_quotes(s):
+    if s.count('"""') % 2 == 1:
+        open_triple = '"""'
+    elif s.count("'''") % 2 == 1:
+        open_triple = "'''"
```

Comment:
Maybe I misunderstand the use of this function. I am wondering if there is a logic issue here ... If you pass the a string with this content: `''' I like """ quotes '''` 

Would it return `"""` even thus the internal quotes should not count.

Or maybe such a string is not possible with Python and the inner quotes must be escaped